### ISSUE TITLE
GpuImages are RasterImages

### ISSUE THREAD

**Issue by davidhesselbom** [cogneco/ooc-kean#514]
`GpuImage` and `RasterImage` have things in common that, say, a `VectorImage` would not. Shouldn't there be a `CpuImage` class, and shouldn't both `CpuImage` and `GpuImage` inherit from `RasterImage`, and `RasterImage` from `Image`, instead of, as it is now, `GpuImage` and `RasterImage` from `Image`?

In other words, currently:
- `Image`
    - `GpuImage`
    - `RasterImage`

Suggestion:
- `Image`
    - `RasterImage`
        - `CpuImage`
        - `GpuImage`
    - `VectorImage` (doesn't exist yet, but it might, at some point)

**Comment by marcusnaslund** [comment]
@sebastianbaginski - thoughts?

**Comment by sebastianbaginski** [comment]
Leave it open, it is still valid.

**Comment by marcusnaslund** [comment]
Still valid @emilwestergren ?

**Comment by davidpiuva** [comment]
I suggest just renaming `rasterImage` to `cpuImage` since we are trying to reduce the number of abstraction layers. If we ever have a vector image type, we can add that later.

**Comment by emilwestergren** [comment]
Still valid, and I agree with @davidpiuva

**Comment by marcusnaslund** [comment]
Sounds like a simple fix then. Everything named `Raster*` is renamed to `Cpu*`?

**Comment by davidpiuva** [comment]
Yes, with lots of version conflicts for anyone doing anything at the same time. Can be done when most are on vacation.

**Comment by emilwestergren** [comment]
It's a simple fix but not prioritized since it will mess things up with little gain. @davidpiuva sounds good

**Comment by marcusnaslund** [comment]
Excellent. I just want issues discussed so they're easier to fix when the time comes.
